include/exec/any_sender_of.hpp

```diff
@@ -0,0 +1,805 @@
+        template <class _Tag, class... _As>
+            requires __callable<const __query_vtable<_SchedulerQueries>&, _Tag, void*, _As...>
+          friend auto tag_invoke(_Tag, const __scheduler& __self, _As&&... __as)
+            noexcept(__nothrow_callable<const __query_vtable<_SchedulerQueries>&, _Tag, void*, _As...>)
+            -> __call_result_t<const __query_vtable<_SchedulerQueries>&, _Tag, void*, _As...> {
+            return __get_vtable(__self.__storage_)
+                ->__queries()(_Tag{}, __get_object_pointer(__self.__storage_), (_As&&) __as...);
+          }
+
+        friend bool operator==(const __scheduler& __self, const __scheduler& __other) noexcept {
+          if (__get_vtable(__self.__storage_) != __get_vtable(__other.__storage_)) {
+            return false;
+          }
+          void* __p = __get_object_pointer(__self.__storage_);
+          void* __o = __get_object_pointer(__other.__storage_);
+                  // if both object pointers are not null, use the virtual equal_to function
+          return (__p && __o && __get_vtable(__self.__storage_)->__equal_to_(__p, __o)) 
+                  // if both object pointers are nullptrs, they are equal
+                || (!__p && !__o);
+        }
+
+        friend bool operator!=(const __scheduler& __self, const __scheduler& __other) noexcept {
+          return !(__self == __other);
+        }
+
+        __copyable_storage_t<__vtable> __storage_{};
+      };
+
+    template <class _Sigs, class _OldSQs, class _RQ, class... _NewSQs>
+      auto __with_sender_queries_fn(__sender<_Sigs, _OldSQs, _RQ>*, _NewSQs*...) 
+        -> __sender<_Sigs, __minvoke<__mconcat<>, _OldSQs, __types<_NewSQs...>>, _RQ>;
+
+    template <class _Sigs, class _SchQs, class _RQ, class _OldSQs, class... _NewSQs>
+      auto __with_sender_queries_fn(__scheduler<_Sigs, _SchQs, _OldSQs, _RQ>*, _NewSQs*...) 
+        -> __scheduler<_Sigs, _SchQs, __minvoke<__mconcat<>, _OldSQs, __types<_NewSQs...>>, _RQ>;
+
+    template <class _S, class... _Queries>
+      using with_sender_queries = decltype(__with_sender_queries_fn((_S*)nullptr, (_Queries*)nullptr...));
+
+    template <class _Sigs, class _SQs, class _OldRQs, class... _NewRQs>
+      auto __with_receiver_queries_fn(__sender<_Sigs, _SQs, _OldRQs>*, _NewRQs*...) 
+        -> __sender<_Sigs, _SQs, __minvoke<__mconcat<>, _OldRQs, __types<_NewRQs...>>>;
+
+    template <class _S, class... _Queries>
+      using with_receiver_queries = decltype(__with_receiver_queries_fn((_S*)nullptr, (_Queries*)nullptr...));
+
+    template <class _Sigs, class _SchSQs, class _RQ, class _SQs, class... _NewSchSQs>
+      auto __with_scheduler_queries_fn(__scheduler<_Sigs, _SchSQs, _SQs, _RQ>*, _NewSchSQs*...) 
+        -> __scheduler<_Sigs, __minvoke<__mconcat<>, _SchSQs, __types<_NewSchSQs...>>, _SQs, _RQ>;
+
+    template <class _S, class... _Queries>
+      using with_scheduler_queries = decltype(__with_scheduler_queries_fn((_S*)nullptr, (_Queries*)nullptr...));
```

Comment:
I see what you're doing here. I like the direction. Needs tests, obvs.

Another potential direction I've noodled on is to have folks name a type-erased receiver type w/ queries, use that to name a type-erased sender w/ queries, and then use that to name a type-erased scheduler w/ queries.

I'm not sure yet which one I would like better.